packages/radio-group/src/Radio.tsx

```diff
@@ -295,16 +294,19 @@ export type RadioProps = Omit<React.ComponentPropsWithoutRef<'input'>, 'size'> &
 function Radio({
   children,
   className,
-  onChange,
+  onChange = () => {},
   value,
   disabled,
   id,
   name,
   darkMode,
-  checked = false,
+  checked,
```

Comment:
Removing default here, such that we're able to check for the existence of the `checked` prop when deciding if the component is controlled or uncontrolled.